File: Scala/src/test/scala/org/revian/MyStackTest.scala
package org.revian

import org.scalatest.funsuite.AnyFunSuite

class MyStackTest extends AnyFunSuite {

  val myStack = new MyStack

  test("Push element onto stack") {
    myStack.push(10)
    assert(myStack.size() == 1)
  }

  test("Pop element from stack") {
    myStack.push(10)
    assert(myStack.pop() == 10)
    assert(myStack.size() == 0)
  }

  test("Pop from empty stack should throw exception") {
    assertThrows[IllegalStateException] {
      myStack.pop()
    }
  }
}

File: Java/src/test/java/org/revian/MyStackTest.java
package org.revian;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class MyStackTest {
    private MyStack stack;

    @BeforeEach
    void setUp() {
        stack = new MyStack();
    }

    @Test
    void testPushAndPop() {
        stack.push(1);
        stack.push(2);
        assertEquals(2, stack.pop());
        assertEquals(1, stack.pop());
    }

    @Test
    void testPeek() {
        stack.push(1);
        assertEquals(1, stack.peek());
    }

    @Test
    void testPopEmptyStack() {
        assertThrows(IllegalStateException.class, () -> stack.pop());
    }

    @Test
    void testPeekEmptyStack() {
        assertThrows(IllegalStateException.class, () -> stack.peek());
    }
}
